client/lib/client.dart

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import 'dart:math';
+
+import 'package:firebase/firebase_io.dart' as fb;
+import 'package:protobuf/protobuf.dart';
+import 'package:spine_client/google/protobuf/empty.pb.dart';
+import 'package:spine_client/rest_firebase_client.dart';
+import 'package:spine_client/spine/core/ack.pb.dart';
+import 'package:spine_client/spine/core/response.pb.dart';
+import 'package:spine_client/spine_client.dart';
+
+import 'blog/blog.pb.dart';
+import 'blog/identifiers.pb.dart';
+import 'types.dart' as blogTypes;
+
+/// A client which executes queries and sends commands.
+abstract class Client {
+
+    /// Fetches all the `Blog`s.
+    Stream<Blog> fetchBlogs();
+
+    /// Fetches a single `Blog` with all of its `Post`s.
+    Future<BlogView> fetchBlogWithPosts(BlogId id);
+
+    /// Posts the given command message.
+    Future<Ack> post(GeneratedMessage command);
+}
+
+/// A backend client which sends queries and commands over the network to a real server.
+class NetworkClient extends Client {
+
+    static final _actor = UserId()
+        ..value = 'Example Dart client'
+        ..freeze();
+
+    final BackendClient _backend;
+    final ActorRequestFactory _factory;
```

Comment:
We can do that. I suggest we set up this example first in case we need to use the "low-level" client API before the "high-level" one will be implemented.

Java client and other clients are in an eternal tortoise-hare relationship. The only way of ending it is some sort of a multiplatform implementation, e.g. with Kotin. We may want to look into it, but I'm not certain that it's the right way to go.